editoast/editoast_models/src/lib.rs

```diff
@@ -1,11 +1,9 @@
 use diesel_async::{pooled_connection::deadpool::Pool, AsyncPgConnection};
 
-mod db_connection_pool;
-mod error;
+pub mod db_connection_pool;
 
-pub use db_connection_pool::create_connection_pool;
+pub use db_connection_pool::DatabasePoolError;
 pub use db_connection_pool::DbConnectionPoolV2;
```

Comment:
I'm not sure that I understand the logic about what should be publicly exposed from `lib.rs` and what should stay namespaced in `db_connection_pool`. If you make `db_connection_pool` public, then maybe it's not necessary to make public any of the underlying object?

For example, here, only `DatabasePoolError` is exposed... but in the diff of `editoast/src/error.rs`, there is the following code
```rust

use editoast_models::db_connection_pool::DatabasePoolBuildError;
use editoast_models::db_connection_pool::DatabasePoolError;
```

Reply:
My logic here was that since `DbConnectionPoolV2` is one of the most commonly used type of `models`, it's convenient to expose it at the root of the package. Likewise `create_connection_pool` won't be used often, so it's fine to ask its caller to import the full module path.

`pub use DatabasePoolError` was a bad last minute change, thanks for pointing it out 😬